pkg/kubelet/kubelet.go

```diff
@@ -1963,19 +1966,34 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 		}
 		klog.V(4).Infof("SyncLoop (SYNC): %d pods; %s", len(podsToSync), format.Pods(podsToSync))
 		handler.HandlePodSyncs(podsToSync)
-	case update := <-kl.livenessManager.Updates():
-		if update.Result == proberesults.Failure {
+	case updateLiveness := <-kl.livenessManager.Updates():
+		if updateLiveness.Result == proberesults.Failure {
 			// The liveness manager detected a failure; sync the pod.
 
 			// We should not use the pod from livenessManager, because it is never updated after
 			// initialization.
-			pod, ok := kl.podManager.GetPodByUID(update.PodUID)
+			pod, ok := kl.podManager.GetPodByUID(updateLiveness.PodUID)
 			if !ok {
 				// If the pod no longer exists, ignore the update.
-				klog.V(4).Infof("SyncLoop (container unhealthy): ignore irrelevant update: %#v", update)
+				klog.V(4).Infof("SyncLoop (container unhealthy - liveness): ignore irrelevant update: %#v", updateLiveness)
 				break
 			}
-			klog.V(1).Infof("SyncLoop (container unhealthy): %q", format.Pod(pod))
+			klog.V(1).Infof("SyncLoop (container unhealthy - liveness): %q", format.Pod(pod))
+			handler.HandlePodSyncs([]*v1.Pod{pod})
+		}
+	case updateStartup := <-kl.startupManager.Updates():
```

Comment:
I think this will "race" (not in the sense of a memory race, but the fact that only one of them will get updates to the channel) against this update mechanism, and ignore Success results:
https://github.com/kubernetes/kubernetes/blob/323f99ea8c4faf44090b7554fbfaad2d25120b65/pkg/kubelet/prober/prober_manager.go#L305-L310

Not sure if that is intended or not..

Reply:
But the channel isn't shared... one listens from `startupManager.Updates` and the other from `livenessManager.Updates`. (maybe I don't get the point)

Reply:
I hope so... but I have never really worked on that code...